### DISCUSSION TITLE
Help to understand this error code

### DISCUSSION THREAD

**Discussion by Arinomo** [OpenModelica/OpenModelica#11869] [Q&A] [answered]
I simulate model of a system and got following error:

```
Simulation process failed. Exited with code 0xffffffffffffffff.
Fatal error on line 21, column 10: Opening and ending tag mismatch.
C:/Users/<redacted>/AppData/Local/Temp/OpenModelica/OMEdit/ZentralRechner_2/ZentralRechner_2.exe -port=57483 -logFormat=xmltcp -override=startTime=0,stopTime=100,stepSize=0.001,tolerance=1e-6,solver=euler,outputFormat=mat,variableFilter=.* -r=C:/Users/<redacted>/AppData/Local/Temp/OpenModelica/OMEdit/ZentralRechner_2/ZentralRechner_2_res.mat -w -lv=LOG_STDOUT,LOG_ASSERT,LOG_STATS -inputPath=C:/Users/<redacted>/AppData/Local/Temp/OpenModelica/OMEdit/ZentralRechner_2 -outputPath=C:/Users/<redacted>/AppData/Local/Temp/OpenModelica/OMEdit/ZentralRechner_2
... loading "Eta" from "T:\<redacted>\BuckConverterEfficiency.txt" 
... loading "Profil1" from "R:/9010_Misc/<redacted>.txt" 
The initialization finished successfully with 3 homotopy steps.
Homotopy solver Newton iteration: Maximum number of iterations reached at time 0.001000, but no root found.
Error solving nonlinear system 211 at time 0.001
Error solving nonlinear system 211 at time 0.001
nonlinear system 211 fails: at t=0.001
[1] Real front.limiter.u(start=0, nominal=1)
```
I'm having a hard time to understand what causing the error or why.

`Fatal error on line 21, column 10: Opening and ending tag mismatch.`:
- Which line was the message referring to? the simulated model has only 31 lines, with connection assigment in line 21 `connect(constantVoltage.p, dcdc1.Input) annotation(
    Line(points = {{-70, -60}, {-94, -60}, {-94, -30}, {-90, -30}}, color = {0, 0, 255}));`
- which tag mismatch is causing the error? isn't class opening and ending was checked when translating?

`nonlinear system 211`:
![image](https://github.com/OpenModelica/OpenModelica/assets/52284681/2371acd3-4d3c-4702-9782-7080259c4fda)

 `[1] Real front.limiter.u(start=0, nominal=1)`:

- `u`  is an input of standard MSL Block `Blocks.Nonlinear.Limiter` and connected to an output of `Modelica.Blocks.Continuous.PI`

The model works with `dassl` and `cvode` solver, but i want to integrate this model in HiL and the the moment only `euler` can be successfully integrated in my tool-chain.

Is it just a limitation of `euler`?

**Comment by casella** [comment, marked as answer]
Absolutely. The model is probably very stiff, so after one step of explicit Euler some state values go through the roof and cause all kind of trouble. Try with smaller time steps, if you need to reduce them too much to achieve stability, you really need an implicit solver. Or, to change the model to eliminate the stiffness.

**Reply by Arinomo** [reply]
@casella thanks for the answer. it was indeed cause by the model that contain `Limiter` block. I replaced the model and it works ok with `euler`.

But does it explain why OM suddenly throw a tag mismatch error? Because from my perspective, it looks like the error was triggered by an error in source file, either written by the user or in the library that the user used.

this part of the error message was actually good at hinting it was solver problem and not modeling problem
```
Homotopy solver Newton iteration: Maximum number of iterations reached at time 0.001000, but no root found.
Error solving nonlinear system 211 at time 0.001
Error solving nonlinear system 211 at time 0.001
nonlinear system 211 fails: at t=0.001
[1] Real front.limiter.u(start=0, nominal=1)

```

**Reply by casella** [reply]
> But does it explain why OM suddenly throw a tag mismatch error?

I'm not sure, but I understand this happens when the simulation aborts prematurely, probably some files are not updated correctly. Anyway, that was not the root cause.
